File: tests/register_test.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time
import random
import string

def generate_random_email():
    random_str = ''.join(random.choices(string.ascii_lowercase + string.digits, k=8))
    return f"testuser_{random_str}@gmail.com"

def test_register():
    print("Starting browser...")
    driver = webdriver.Chrome()
    
    try:
        print("Navigating to signup page...")
        driver.get("http://localhost:3000/signup")
        driver.maximize_window()
        
        wait = WebDriverWait(driver, 10)
        
        # Locate inputs
        name_input = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, "input[placeholder='Your name']")))
        email_input = driver.find_element(By.CSS_SELECTOR, "input[type='email']")
        password_input = driver.find_element(By.CSS_SELECTOR, "input[placeholder='e.g. Pass@123']")
        confirm_input = driver.find_element(By.CSS_SELECTOR, "input[placeholder='Re-type password']")
        submit_button = driver.find_element(By.CSS_SELECTOR, "button[type='submit']")
        
        test_email = generate_random_email()
        test_password = "TestPassword@123"
        
        print(f"Registering new user: {test_email}")
        name_input.send_keys("Selenium Test User")
        email_input.send_keys(test_email)
        password_input.send_keys(test_password)
        confirm_input.send_keys(test_password)
        
        print("Submitting registration...")
        driver.execute_script("arguments[0].click();", submit_button)
        
        # Verify success by waiting for dashboard redirect
        wait.until(EC.url_contains("/dashboard"))
        print("\n✅ TEST PASSED: Successfully registered and redirected to the Dashboard!")
        
    except Exception as e:
        print(f"\n❌ TEST FAILED: {str(e)}")
        
    finally:
        time.sleep(3)
        print("Closing browser...")
        driver.quit()

if __name__ == "__main__":
    test_register()

File: tests/register_test.js
const { Builder, By, until } = require('selenium-webdriver');

async function testRegister() {
  console.log("Starting browser...");
  let driver = await new Builder().forBrowser('chrome').build();

  try {
    console.log("Navigating to signup page...");
    await driver.get('http://localhost:3000/signup');
    await driver.manage().window().maximize();

    const nameInput = await driver.wait(until.elementLocated(By.css("input[placeholder='Your name']")), 10000);
    const emailInput = await driver.findElement(By.css("input[type='email']"));
    const passwordInput = await driver.findElement(By.css("input[placeholder='e.g. Pass@123']"));
    const confirmInput = await driver.findElement(By.css("input[placeholder='Re-type password']"));
    const submitBtn = await driver.findElement(By.css("button[type='submit']"));

    const randomStr = Math.random().toString(36).substring(2, 10);
    const testEmail = `testuser_${randomStr}@gmail.com`;
    const testPassword = "TestPassword@123";

    console.log(`Registering new user: ${testEmail}`);
    await nameInput.sendKeys("Selenium Test User JS");
    await emailInput.sendKeys(testEmail);
    await passwordInput.sendKeys(testPassword);
    await confirmInput.sendKeys(testPassword);

    console.log("Submitting registration...");
    await driver.executeScript("arguments[0].click();", submitBtn);

    await driver.wait(until.urlContains('/dashboard'), 5000);
    console.log("\n✅ TEST PASSED: Successfully registered and redirected to the Dashboard!");

  } catch (error) {
    console.log(`\n❌ TEST FAILED: ${error.message}`);
  } finally {
    await driver.sleep(3000);
    console.log("Closing browser...");
    await driver.quit();
  }
}

testRegister();
